qa/restart-upgrade/src/test/java/org/opensearch/neuralsearch/bwc/HybridSearchIT.java

```diff
@@ -126,7 +129,7 @@ private HybridQueryBuilder getQueryBuilder(
         neuralQueryBuilder.modelId(modelId);
         neuralQueryBuilder.queryText(QUERY);
         neuralQueryBuilder.k(5);
-        if (expandNestedDocs != null) {
+        if (isClusterOnOrAfterMinReqVersion(EXPAND_NESTED_FIELD.getPreferredName()) && expandNestedDocs != null) {
```

Comment:
I think we should not check cluster version here. methodParameters and rescoreContext does not check cluster version as well.

Reply:
Yeah. Seems like the build will handle this check.

Reply:
Try again with old code. The previous error was because the parameter is provided as true when the field was not nested field. With parameter value False, it should pass. 
```
"Invalid value provided for the [expand_nested_docs] field. [expand_nested_docs] is only supported with a nested field.
```

Reply:
I mean for 2.12 we even should not use this parameter in neuralQueryBuilder.